File: src/backend/python/app/models/todo_item.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class TodoItem(Base):
    """ORM model for a todo item (analogous to an EF Core entity)."""

    __tablename__ = "todo_items"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True, autoincrement=True)
    title: Mapped[str] = mapped_column(String(200), nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    is_completed: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), onupdate=func.now(), nullable=True)

    def __repr__(self) -> str:
        return f"<TodoItem id={self.id!r} title={self.title!r} is_completed={self.is_completed!r}>"

File: src/backend/php/app/Models/TodoItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * Eloquent model for a todo item.
 * Analogous to an EF Core entity class with data annotations.
 *
 * @property int         $id
 * @property string      $title
 * @property string|null $description
 * @property bool        $is_completed
 * @property \Carbon\Carbon $created_at
 * @property \Carbon\Carbon|null $updated_at
 */
class TodoItem extends Model
{
    use HasFactory;

    protected $table = 'todo_items';

    /**
     * Mass-assignable attributes (analogous to [BindProperty] or DTO mapping).
     */
    protected $fillable = [
        'title',
        'description',
        'is_completed',
        'created_by_user_id',
        'updated_by_user_id',
    ];

    /**
     * Default attribute values (mirrors the DB column defaults).
     */
    protected $attributes = [
        'is_completed' => false,
    ];

    /**
     * Attribute type casts (analogous to EF value converters / column types).
     */
    protected $casts = [
        'is_completed' => 'boolean',
        'created_at'   => 'datetime',
        'updated_at'   => 'datetime',
    ];

    public function attachments(): HasMany
    {
        return $this->hasMany(TodoItemAttachment::class);
    }
}
